package.json

```diff
@@ -60,14 +60,23 @@
     "chai": "^4.2.0",
     "chai-as-promised": "^7.1.1",
     "husky": "^1.1.2",
+    "lint-staged": "^8.0.4",
     "mocha": "^5.2.0",
     "nock": "^10.0.1",
     "openvpn-client": "0.0.2",
     "resin-lint": "^2.0.1"
   },
   "husky": {
     "hooks": {
+      "pre-commit": "lint-staged",
       "pre-push": "npm run lint"
     }
+  },
+  "lint-staged": {
+    "*.ts": [
+      "prettier --config ./node_modules/resin-lint/config/.prettierrc --write",
+      "resin-lint --typescript --no-prettier",
+      "git add"
```

Comment:
Heads up — This will prevent you from committing partially staged files — it’ll stage the whole file.

Reply:
@dfunckt I believe `lint-staged` will stash any changes to files which also have staged changes during this process.